### PR TITLE
Feat watch later

### PR DESCRIPTION
**Pull request by Kothalkarpriya**
Created a feature for video to be added to watch later page. 
Created a single video page. 
On the video listing page, if we click on the video card we will be redirected to the single video page and from there we can add and remove videos into watch later and remove them from it too. Similarly, on the watch later page, you will see videos if you have added any to it, and from there by clicking the button watch now, you will get back to the single video page.

### CODE REVIEW

File: src/assests/styles/video-card-page.css [hunk 1]
```diff
@@ -0,0 +1,25 @@
+@import "./colors.css";
+
+section {
+    color: white;
+}
+
+.video-item {
+    width: 90%;
+    height: 400px;
```
Comment (harshalini) [inline src/assests/styles/video-card-page.css]: Use can use rem here

File: src/assests/styles/video-listing.css [hunk 2]
```diff
@@ -11,6 +11,7 @@
     padding:10px;
     margin:1rem;
     box-shadow: var(--all-side-shadow);
+    max-width: fit-content;
 }
 .video .container{
     width:250px;
```
Comment (harshalini) [inline src/assests/styles/video-listing.css]: Here also

File: src/context/context.jsx [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
 import { VideoCategoryContext, useVideoCategory } from "./VideoCategoryContext";
+import { WatchLaterContextPro, useWatchLater } from "./WatchLaterContext";
 
 export {
   FilterContextProvider,
   VideoCategoryContext,
   useFilter,
   useVideoCategory,
+  WatchLaterContextPro,
+  useWatchLater,
```
Comment (harshalini) [inline src/context/context.jsx:11]: I like that you have made a separate file to export

File: src/pages/pages.jsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import Home from "./Home/Home";
 import Playlist from "./Playlist/Playlist";
 import VideoListing from "./VideoListing/VideoListing";
+import VideoCardPage from "./VideoListing/VideoCardPage";
```
Comment (harshalini) [inline src/pages/pages.jsx:4]: For these also you can make a separate file to export from there

**Review by harshalini — APPROVED** [review]
Overall code is well structured and readable

File: src/assests/styles/video-card-page.css (hunk 1 above)
Comment (Kothalkarpriya) [reply src/assests/styles/video-card-page.css]: I will modify it

File: src/assests/styles/video-listing.css (hunk 2 above)
Comment (Kothalkarpriya) [reply src/assests/styles/video-listing.css]: Yes Thanks

File: src/context/context.jsx (hunk 3 above)
Comment (Kothalkarpriya) [reply src/context/context.jsx:11]: Thanks

File: src/pages/pages.jsx (hunk 4 above)
Comment (Kothalkarpriya) [reply src/pages/pages.jsx:4]: It is already a separate file Which is importing and exporting

File: src/components/AddToWatchBtns.jsx [hunk 5]
```diff
@@ -1,15 +1,53 @@
-// import { BsThreeDotsVertical } from "react-icons/bs";
 import { AiFillLike, AiOutlineShareAlt } from "react-icons/ai";
 import { MdPlaylistAdd, MdOutlineWatchLater } from "react-icons/md";
```
Comment (dikshadhote) [inline src/components/AddToWatchBtns.jsx:2]: Can include all icons in one file

File: src/route/RoutePath.jsx [hunk 6]
```diff
@@ -24,6 +25,7 @@ export default function RoutePath() {
       <Route path="/Logout" element={<Logout />} />
       <Route path="/Favourites" element={<Favourites />} />
       <Route path="/WatchLater" element={<WatchLater />} />
```
Comment (dikshadhote) [inline src/route/RoutePath.jsx:27]: Favourite and watch later should be private route

**Review by dikshadhote — APPROVED** [review]
Overall code is well structured. But remove from watchlist functioanlity should be available in watch list page, instead of explore.

File: src/components/AddToWatchBtns.jsx (hunk 5 above)
Comment (Kothalkarpriya) [reply src/components/AddToWatchBtns.jsx:2]: ok

File: src/route/RoutePath.jsx (hunk 6 above)
Comment (Kothalkarpriya) [reply src/route/RoutePath.jsx:27]: They are going to be implemented after the p0 screens are implemented

**Comment by Kothalkarpriya** [comment]
> Overall code is well structured. But remove from watchlist functioanlity should be available in watch list page, instead of explore.

Added remove button in Watch later

**Pull request merged**